controllers/stargate/stargate_controller.go

```diff
@@ -265,6 +265,16 @@ func (r *StargateReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		}
 	}
 
+	// create Vector configmap
+	if vectorReconcileResult, err := r.reconcileVector(ctx, *stargate, actualDc, r.Client, logger); err != nil {
+		return vectorReconcileResult, err
+	} else if vectorReconcileResult.Requeue {
+		return vectorReconcileResult, nil
+	}
+
+	// inject Vector
+	injectVectorAgentForStargate(stargate, desiredDeployments, actualDc.Name, actualDc.Spec.ClusterName, logger)
```

Comment:
You need to call this function from inside `stargateutil.NewDeployments` because the hash annotation is being computed there.

Reply:
Yes, you are correct. That is what was causing my changes to not get picked up.